### PR TITLE
Dmitry Goman - React Task 1

### PR DESCRIPTION
**Pull request by dima-red**

### CODE REVIEW

File: src/components/blocks/Header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import Logo from 'components/reusable/Logo/Logo';
+import { IconButton } from '@material-ui/core';
+import HomeIcon from '@material-ui/icons/Home';
+import Avatar from '@material-ui/core/Avatar';
+import { header, headerContent, divider } from './Header.scss';
+import { avatar } from '../../reusable/Avatar/Avatar.scss';
+import avatartSrc from 'assets/images/DG.png';
+
+const logoProps = {
+  text: 'D&G (Dima & Goman)'
```
Comment (AV29) [inline src/components/blocks/Header/Header.jsx]: Hulk & Bruce Banner :)

File: src/components/blocks/Header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <header className={ header }>
+      <div className={ headerContent }>
+        <IconButton >
+          <HomeIcon fontSize="large"/>
+        </IconButton>
+        <div className={ divider }></div>
+        <Logo logoContent={ logoProps }/>
+      </div>
+      <div className={ avatar }>
+        <Avatar alt="avatar" src={ avatartSrc } />
```
Comment (AV29) [inline src/components/blocks/Header/Header.jsx]: Could this Avatar (svg) component take className property? if so - then <div className={ avatar }> is redundant

File: src/components/blocks/Main/Main.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { mainContent, mainWrapper, cityItemWrapper } from './Main.scss';
+
+const Main = (props) => {
+    const { cityWeatherReducer } = props.cityData;
+
+    return (
+        <main className={ mainWrapper }>
+            <Switch>
+                <Route exact path="/">
+                    <Paper className={ mainContent }>
```
Comment (AV29) [inline src/components/blocks/Main/Main.jsx]: Component Paper... what does it do ? I personally don't get it ...

File: src/components/blocks/SideBar/SideBar.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.leftGrid {
+    // min-width: 50px;
```
Comment (AV29) [inline src/components/blocks/SideBar/SideBar.scss]: IMHO, let's not commit comments

File: src/components/pages/CurrentCity/CurrentCity.jsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Button } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import { Link } from "react-router-dom";
+import WeatherIcon from '../../reusable/WeatherIcon/WeatherIcon';
```
Comment (AV29) [inline src/components/pages/CurrentCity/CurrentCity.jsx]: this import should be on line 8 instead

File: src/components/pages/CurrentCity/CurrentCity.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    date,
+    time,
+    city,
+    temperature
+} from './CurrentCity.scss';
+
+class CurrentCity extends React.Component {
+    constructor(props) {
+        super(props);
+        this.cityData = this.props.cityData;
```
Comment (AV29) [inline src/components/pages/CurrentCity/CurrentCity.jsx]: the problem you may face with is that when props change - your component (CurrentCity) won't reflect those changes. Constructor executes only once per Component's lifecycle setting props data to **this**. Then you are using instance (this) data do reflect it on a view. You may never notice in your specific ad-hoc case: say, your CurrentCity component being rerendered from the scratch every time executing constructor each time. But this kind of approach is bug prone.

Instead , why not just render data from props directly ?

File: src/components/pages/CurrentCity/CurrentCity.jsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+                    </Grid>
+                </Grid>
+                <div className={ editLocationButtonWrapper }>
+                    <Link to="/cities">
+                        <Button
+                            size="large"
+                            variant="contained"
+                            color="default"
+                            id={ editLocationButton }
+                            startIcon={ <EditLocationIcon fontSize="large"/> }
```
Comment (AV29) [inline src/components/pages/CurrentCity/CurrentCity.jsx]: what does startIcon property mean ? 

Could this Button  and IconButton be merged somehow into one component ?

File: src/components/reusable/CityItem/CityItem.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Grid from '@material-ui/core/Grid';
+import WeatherIcon from '../../reusable/WeatherIcon/WeatherIcon';
```
Comment (AV29) [inline src/components/reusable/CityItem/CityItem.jsx]: How about move WeatherIcon import on line 6

File: src/components/reusable/CityItem/CityItem.jsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import { Link } from "react-router-dom";
+import {
```
Comment (AV29) [inline src/components/reusable/CityItem/CityItem.jsx]: how about just **import styles from './CityItem.scss'** ? ...and place it at the end of import block

File: src/components/reusable/CityItem/CityItem.jsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    windWrapper,
+    gridItem,
+    windText,
+    city,
+    temperature
+} from './CityItem.scss';
+import { setCurrentCity } from '../../../actions/cityActions';
+
+class CityItem extends React.Component {
+    cityData = null;
```
Comment (AV29) [inline src/components/reusable/CityItem/CityItem.jsx]: I don't think it is necessary at all here

File: src/components/reusable/CityItem/CityItem.jsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+    constructor(props) {
+        super(props);
+        this.cityData = this.props.cityData;
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick() {
+        this.cityData.isCurrentCity = true;
```
Comment (AV29) [inline src/components/reusable/CityItem/CityItem.jsx]: be carefull with mutations... they could become confusing not only for you...

You should just track it via redux. You are already dispatching an action to setCurrent city. Just change respective flag via reducer.

File: src/components/reusable/CurrentDate/CurrentDate.jsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+class CurrentDate extends React.Component {
+    today = null;
```
Comment (AV29) [inline src/components/reusable/CurrentDate/CurrentDate.jsx]: you don't have to declare these 

you are already assigning them in c-tor

File: src/components/reusable/CurrentDate/CurrentDate.jsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    date = null;
+
+    constructor() {
+        super();
+        this.today = new Date();
```
Comment (AV29) [inline src/components/reusable/CurrentDate/CurrentDate.jsx]: What if you run you App at 11 59: 59 PM ?  within a second new date should be rendered but I doubt that would happen in your case... ;)

File: src/components/reusable/WeatherIcon/WeatherIcon.jsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+class WeatherIcon extends React.Component {
+    weatherIcons = null;
+
+    constructor(props) {
+        super(props);
+        const iconInfo = this.props.iconInfo;
+        const isCurrentCity = this.props.isCurrentCity;
+        this.weather = iconInfo.weather;
+        this.iconType = iconInfo.iconType;
+        this.width = isCurrentCity ? ICON_WIDTH_BIG : ICON_WIDTH_SMALL;
```
Comment (AV29) [inline src/components/reusable/WeatherIcon/WeatherIcon.jsx]: Can you elaborate why you don;t use prop values directly but instead store them on instance in construcotr?

File: src/reducers/cityWeatherReducer.jsx [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+    allCities: [
+        {
+            weather: "clear-night",
+            iconType: "CLEAR_NIGHT",
+            cityName: "Moscow",
+            temperature: "+5°C",
+            wind: "SE",
+            id: 1,
+        },
+
```
Comment (AV29) [inline src/reducers/cityWeatherReducer.jsx]: redundant blank lines ?

File: src/reducers/cityWeatherReducer.jsx [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+            cityName: "Milan",
+            temperature: "+17°C",
+            wind: "NW",
+            id: 4,
+        } 
+    ]
+};
+
+const cityWeatherReducer = (state = initialState, action) => {
+    console.log("Reducer");
```
Comment (AV29) [inline src/reducers/cityWeatherReducer.jsx]: don't forget to remove console.logs ;)

File: src/containers/MainContainer.jsx [hunk 17]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Main from '../components/blocks/Main/Main';
+
+const MainContainer = (props) => (
```
[the remark this answers is not in the document]
Comment (dima-red) [reply src/containers/MainContainer.jsx]: fixed

File: src/components/blocks/Header/Header.jsx (hunk 2 above)
Comment (dima-red) [reply src/components/blocks/Header/Header.jsx]: It can, but I need a wrapper for better display.

**Pull request closed without merging**